api/src/codec-to-json.ts

```diff
@@ -0,0 +1,47 @@
+import { bytes, codec, context } from "@typeberry/block";
+import type { FastifyReply, FastifyRequest } from "fastify";
+import { createErrorResponse } from "./error";
+import { kinds } from "./kinds";
+
+type Req = { Params: { id: string }; Body: { codec: string } };
+type SuccessRes = { data: { json: string } };
+type Res = { Body: SuccessRes };
+
+export function codecToJson(req: FastifyRequest<Req>, reply: FastifyReply<Res>) {
+  const descriptorId = req.params.id;
```

Comment:
Is there any validation here? Can `req.params` be `undefined`?

Reply:
it is required in `api.yaml`